tree/ntuple/v7/inc/ROOT/RNTuple.hxx

```diff
@@ -458,6 +465,67 @@ public:
    ClusterSize_t *GetOffsetPtr() { return &fOffset; }
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RNTuple
+\ingroup NTuple
+\brief Representation of an RNTuple data set in a ROOT file
+
+This class provides an API entry point to an RNTuple stored in a ROOT file. Its main purpose is to
+construct a page source for an RNTuple, which in turn can be used to read an RNTuple with an RDF or
+an RNTupleReader.
+
+For instance, for an RNTuple called "Events" in a ROOT file, usage can be
+~~~ {.cpp}
+auto f = TFile::Open("data.root");
+auto ntpl = f->Get<ROOT::Experimental::RNTuple>("Events");
+
+auto pageSource = ntpl->MakePageSource();
+// or
+auto reader = RNTupleReader::Open(ntpl);
+~~~
+*/
+// clang-format on
+class RNTuple final {
+   friend class ROOT::Experimental::Internal::RNTupleFileWriter;
+   friend struct ROOT::Experimental::Internal::RNTupleTester;
+
+private:
+   /// Persistent data members need to be identical to the ones from RFileNTupleAnchor
+   /// TODO(jblomer): Remove unneeded fChecksum, fVersion, fSize, fReserved once RNTuple moves out of experimental
+
+   std::int32_t fChecksum = 0;
```

Comment:
Would it be plausible to use Composition or Inheritance from the Anchor to avoid the declaration duplication and reduce the number of copies needed?

Reply:
That was my first approach. The problem is that it makes the `RNTuple` custom streamer more complicated. Because standard ROOT I/O and the minifile I/O should interoperate, the `RNTuple` custom streamer would need to write itself as its `RFileNTupleAnchor` base class while keeping the `RNTuple` class name. It seemed cleaner then to just duplicate the data members (which will eventually be reduced to 6 data members).

Reply:
If/Since there is no persistent data in `RNtuple` that is not in `RNTupleAnchor` and since we already
have a custom streamer, this can be handled via
```
void ROOT::Experimental::RNTuple::Streamer(TBuffer &buf)
{
   static TClassRef RNTupleAnchorClass("ROOT::Experimental::RNTupleAnchor");
   if (buf.IsReading()) {
      RNTupleAnchorClass->ReadBuffer(buf, (RNTupleAnchor*)this); // or &(this->fAnchor)
```
(and you can still give a version number (without ClassDef) by specifying it in the LinkDef file or alternatively by a macro in the source file).

Reply:
Neat! Done.